src/compute/src/render/join/linear_join.rs

```diff
@@ -251,7 +256,7 @@ where
         errors.push(errs);
         let arranged =
             keyed.mz_arrange::<RowSpine<_, _, _, _>>("JoinStage", enable_arrangement_size_logging);
-        joined = JoinedFlavor::Local(arranged);
+        joined = JoinedFlavor::Local(SpecializedArrangement::RowRow(arranged));
```

Comment:
Why can't we specialize this? (Maybe you do so in a future commit; if so, ignore this comment).

Reply:
We can! But to make this work, we need to propagate the computation of the input types for the join stream as you join in more of the relations. And then also specialize the code here with appropriate dispatch functions. I felt this was too much to squeeze in into this one PR, so I'd like to tackle it in a subsequent PR to close #22104.